src/app.test.js

```diff
@@ -180,16 +201,32 @@ test("shows outstanding todos when filter is set to Outstanding", () => {
   expect(queryByDisplayValue(todoList, "Take dog for a walk!")).toBeVisible();
 });
 
-/**
- * SETUP
- * render todo app
- * input user name
- * add few todos
- * destroy todo app
- * ACT
- * re render todo app
- * input same user name
- * ASSERT
- * see previously added todos rendered
- */
-test.todo("renders todo list for existing user");
+test("renders empty todo list when changing to a new user", () => {
+  // setup
+  setUser();
+  const todoList = screen.getByTestId("todo-list");
+  addTodo(todoList, "Take wife for a walk!", true);
+  addTodo(todoList, "Take dog for a walk!");
+
+  // act
+  setUser("Marton");
+
+  // assert
+  expect(todoList).toBeEmptyDOMElement();
+});
+
+test("renders todo list for existing user", () => {
+  // setup
+  setUser("A");
+  const todoList = screen.getByTestId("todo-list");
+  addTodo(todoList, "Todo 1 for user A", true);
+  addTodo(todoList, "Todo 2 for user A");
+
+  // act
+  setUser("B");
```

Comment:
as a safety check I'd add the assertion here as well:

`expect(todoList).toBeEmptyDOMElement();`

then set user "A"